src/sentry/utils/locking/backends/redis.py

```diff
@@ -65,12 +79,18 @@ def get_client(self, key, routing_key=None):
 
 
 class RedisClusterLockBackend(BaseRedisLockBackend):
-    def __init__(self, cluster, prefix="l:", uuid=None):
+    cluster: RedisCluster | StrictRedis
```

Comment:
I wonder if instead of this the baseclass should be generic?  so this would extend `BaseRedisLockBackend[RedisCluster | StrictRedis` ?

Reply:
I don't have any strong opinions on this other than OOP is an overkill and makes things extremely hard for both debugging and optimizing.